src/pages/index.js

```diff
@@ -29,51 +43,124 @@ const renderCard = (item) => {
+);
 
 //cards section
 const cardList = new Section(
   {
-    data: initCards,
-    renderer: renderCard,
+    renderer: () => {
+      api.getInitialCards().then((items) => {
+        items.forEach((item) => {
+          renderCard(item);
+        });
+      });
+    },
   },
   ".page__cards"
 );
 
 //edit form
-const editFormValidator = new FormValidation("profile-edit-form", formData);
+const editFormValidator = new FormValidation("edit", formData);
 editFormValidator.enableValidation();
 
 const editPopup = new PopupWithForm(".modal_type_edit", (values) => {
-  userInformation.setUserInfo(values);
-  editFormValidator.disableSubmit();
-  editPopup.close();
+  editPopup.toggleSaving();
+  api.updateUserInformation(values).then((values) => {
+    userInfo.setUserInfo(values);
+    editPopup.close();
+    editPopup.toggleSaving();
+  });
 });
 
 editButton.addEventListener("click", () => {
-  const newValues = userInformation.getUserInfo();
-  editPopup.setInputValues(Object.values(newValues));
+  const newValues = userInfo.getUserInfo();
+  editPopup.setInputValues([newValues.name, newValues.about]);
+  editFormValidator.resetValidation();
   editPopup.open();
 });
 
 //card form
-const cardFormValidator = new FormValidation("card-add-form", formData);
+const cardFormValidator = new FormValidation("card", formData);
 cardFormValidator.enableValidation();
 
-const cardPopup = new PopupWithForm(".modal_type_card", (values) => {
-  renderCard(values);
-  cardFormValidator.disableSubmit();
-  cardPopup.resetForm();
-  cardPopup.close();
+const cardPopup = new PopupWithForm(".modal_type_card", ({ title, link }) => {
+  const values = { name: title, link: link };
+  cardPopup.toggleSaving();
+  api.postCard(values).then((values) => {
+    renderCard(values);
+    cardFormValidator.disableSubmit();
+    cardPopup.resetForm();
+    cardPopup.close();
+    cardPopup.toggleSaving();
+  });
 });
 
 cardButton.addEventListener("click", () => {
   cardPopup.open();
 });
 
-//preview image
+//edit avatar form
+const avatarFormValidator = new FormValidation("avatar", formData);
+avatarFormValidator.enableValidation();
+
+const editAvatarPopup = new PopupWithForm(
+  ".modal_type_avatar",
+  ({ avatar }) => {
+    editAvatarPopup.toggleSaving();
+    api.updateAvatar({ avatar }).then(({ avatar }) => {
+      userInfo.setAvatar({ avatar });
+      avatarFormValidator.disableSubmit();
+      editAvatarPopup.resetForm();
+      editAvatarPopup.close();
+      editAvatarPopup.toggleSaving();
+    });
+  }
+);
+
+avatarButton.addEventListener("click", () => editAvatarPopup.open());
+
+//preview image popup
 const previewPopup = new PopupWithImage(".modal_type_preview");
 
+//confirmation popup
+const confirmPopup = new PopupWithConfirm(
+  ".modal_type_confirm",
+  ({ deleter, _id }) => {
+    api.deleteCard(_id).then(() => {
+      deleter();
+      confirmPopup.close();
+    });
+  }
+);
+
+// call everything down here <----------
 cardList.clear();
```

Comment:
```js

cardList.clear();
profileSection.renderItems();
cardList.renderItems();
```

You don't need this code


`renderItems` is used for rendering the array of cards only